### PR TITLE
Upgrade rails autoloader version to 6.0

### LINKED ISSUE #2378: Upgrade rails autoloader from version 5.1 to 6.0
We need to upgrade the rails autoloader from version 5.1 to version 6.0
Rails 6 has introduced a new autoloader: **zeitwerk** which is the default autoloader in autoloader version 6

We need to control the version with the following attribute:
`config.load_defaults`

### PR DESCRIPTION
**Pull request by RajeshPaul38**
### Description

Rails 6 has introduced a new autoloader: **zeitwerk** which is the default autoloader in autoloader version 6

### Issues Resolved

#2378 

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by RajeshPaul38** [comment]
Ongoing build: https://buildkite.com/chef/chef-supermarket-main-omnibus-adhoc/builds/273

**Review by saghoshprogress — APPROVED** [review]
LGTM 👍

File: src/supermarket/Gemfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ gem "sidekiq-cron"
 gem "aws-sdk-s3"
 gem "chef", "~> 17.7", require: false
 gem "compass-rails"
-gem "ddtrace", require: false
+gem "ddtrace", "0.53.0", require: false
```
Comment (RajeshPaul38) [inline src/supermarket/Gemfile:19]: pinned to 0.53 because 0.54 has some breaking change

File: src/supermarket/app/lib/supermarket/fips.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 require "openssl"
 
 module Supermarket
-  class FIPS
+  class Fips
```
Comment (RajeshPaul38) [inline src/supermarket/app/lib/supermarket/fips.rb:5]: The new autoloader zeitwerk imposes some strict naming conventions. Hence this change.

File: src/supermarket/db/seeds.rb [hunk 3]
```diff
@@ -84,7 +84,7 @@
       cookbook_version = cookbook.cookbook_versions.where(
         version: "0.#{version_number}.0"
       ).first_or_initialize(
-        description: Faker::Lorem.sentences(1).first,
+        description: Faker::Lorem.sentences(number: 1).first,
```
Comment (RajeshPaul38) [inline src/supermarket/db/seeds.rb:87]: new syntax for faker

**Pull request merged**